cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -179,30 +182,155 @@ def __init__(self, scope: core.Construct, id: str, head_node: HeadNode, **kwargs
+        # EFS FileSystem
+        efs_id = shared_efs.file_system_id.value
+        if not efs_id and shared_efs.mount_dir:
+            efs_resource = efs.CfnFileSystem(
+                scope=self,
+                id=id,
+                encrypted=shared_efs.encrypted.value,
+                kms_key_id=shared_efs.kms_key_id.value,
+                performance_mode=shared_efs.performance_mode.value,
+                provisioned_throughput_in_mibps=shared_efs.provisioned_throughput.value,
+                throughput_mode=shared_efs.throughput_mode.value,
+            )
+            efs_id = efs_resource.ref
+
+        checked_availability_zones = []
+        # Mount Targets for Compute Fleet
+        compute_subnet_ids = self._cluster.compute_subnet_ids
+        for subnet_id in compute_subnet_ids:
+            self._add_efs_mount_target(
+                id, efs_id, self._cluster.compute_security_groups, subnet_id, checked_availability_zones
+            )
+
+        # Mount Target for Head Node
+        self._add_efs_mount_target(
+            id,
+            efs_id,
+            self._cluster.head_node.networking.security_groups.value,
+            self._cluster.head_node.networking.subnet_id.value,
+            checked_availability_zones,
+        )
+        return efs_id
+
+    def _add_efs_mount_target(
+        self, efs_cfn_resource_id, file_system_id, security_groups, subnet_id, checked_availability_zones
+    ):
+        """Create a EFS Mount Point for the file system, if not already available on the same AZ."""
+        availability_zone = AWSApi.instance().ec2.get_availability_zone_of_subnet(subnet_id)
+        if availability_zone not in checked_availability_zones:
+            mount_target_id = AWSApi.instance().efs.get_efs_mount_target_id(file_system_id, availability_zone)
+
+            if not mount_target_id:
+                efs.CfnMountTarget(
+                    scope=self,
+                    id="{0}MT{1}".format(efs_cfn_resource_id, availability_zone),
+                    file_system_id=file_system_id,
+                    security_groups=security_groups,
+                    subnet_id=subnet_id,
+                )
+            checked_availability_zones.append(availability_zone)
+
+    def _add_ebs_volume(self, id: str, shared_ebs: SharedEbs):
+        """Add specific Cfn Resources to map the EBS storage."""
+        ebs_id = shared_ebs.volume_id.value
+        if not ebs_id and shared_ebs.mount_dir:
+            ebs_resource = ec2.CfnVolume(
+                scope=self,
+                id=id,
+                availability_zone=self._cluster.head_node.networking.availability_zone,
+                encrypted=shared_ebs.encrypted.value,
+                iops=shared_ebs.iops.value,
+                throughput=shared_ebs.throughput.value,
+                kms_key_id=shared_ebs.kms_key_id.value,
+                size=shared_ebs.size.value,
+                snapshot_id=shared_ebs.snapshot_id.value,
```

Comment:
Does it works correctly if passing a None value?

Reply:
As discussed, None value should be managed from the lib.